### DISCUSSION TITLE
wayland, or !wayland?

### DISCUSSION THREAD

**Discussion by glyndon** [input-leap/input-leap#1880] [Q&A] [answered]
Q: Is input-leap supposed to be working properly with Wayland at this point, or not?

After searching the Issues and Discussions here, and seeing the phrase "We will also have our eye on Wayland when the time comes" on the main readme, I'm more confused than ever. 
It Seems like many people are using it well with wayland, but it doesn't work on wayland for me.

So I don't know whether I should chase the problem down, or be patient and await full wayland support.

Can someone clarify?
Thanks.

**Comment by shymega** [comment]
For GNOME and the correct XDG portals and supporting libraries - yes.
For wlroots-based WMs/KDE/other compositors - no.

**Reply by glyndon** [reply]
Thanks. I am using GNOME (plain old RaspiOS on a pi5), and all that happens is tooltips popup on the remote display when I move the mouse around there (pointer on remote never visibly moves), and my pointer remains on my screen.

Since it's *supposed* to be working, I guess I've got a problem to puzzle out... :)

**Reply by matdave** [reply]
Gnome implemented a stupid check that you have to accept every time you start the synergy client. However, it does work last I tried. I've been using Waynergy as the client because it black magics past the check. However, it takes a bit of tinkering to get working.

**Reply by shymega** [reply, marked as answer]
> Thanks. I am using GNOME (plain old RaspiOS on a pi5), and all that happens is tooltips popup on the remote display when I move the mouse around there (pointer on remote never visibly moves), and my pointer remains on my screen.
> 
> Since it's _supposed_ to be working, I guess I've got a problem to puzzle out... :)

@glyndon You'll also need the patched `libportal`, `xdg-desktop-portal`, and the DE-specific desktop portal.

I suspect you won't get it working on the Pi just yet. Things have to be backported.

**Reply by shymega** [reply]
> Gnome implemented a stupid check that you have to accept every time you start the synergy client. However, it does work last I tried. I've been using Waynergy as the client because it black magics past the check. However, it takes a bit of tinkering to get working.

@matdave This isn't Synergy, so I'm not sure what you mean. However, Waynergy will not always work with Input Leap, so it's not good to recommend it.

It's part of privacy that the XDG Portal requests permission.
